007_team_5_agent_design/007_team_5_agent_design.tex

```diff
@@ -1 +1,532 @@
+
+It should also be noted that while the Team 5 agent computes favour for agents both above and below its floor, a positive favour value for agents above is helpful to that agent only if they are moved to a floor below the Team 5 agent following a reshuffle: this is because there is no way to reward the behaviour of an agent above you in the tower.
+
+\section{Treaties}\label{sec:team5-treaties}
+The Team 5 agent's treaty strategy consists of three main parts:
+\begin{enumerate}
+    \item Deciding whether to accept or reject proposed treaties
+    \item When to propose a treaty
+    \item What to propose as a treaty
+\end{enumerate}
+
+\subsection*{Responding to Proposed Treaties}\label{sec:team5-treaties-response}
+A proposed treaty is always rejected if:
+\begin{itemize}
+    \item It conflicts with any other active treaties. A treaty is considered to conflict with another treaty if they could ever cause the agent to take incompatible actions. For example, an agent cannot leave more than five units of food and fewer than four units of food at the same time.
+    \item It is badly formed. For example, a treaty which requires the agent to leave more than 100\% of the food on the platform is badly formed.
+    \item It requires agents to leave less than a certain amount of food on the platform, as this is never advantageous to the tower as a whole.
+    \item It requires the agent to leave an exact amount of food on the platform, as this would result in all but one of the agents signed up to the treaty to take no food each day.
+    \item It makes a request based on low HP, low amounts of food on the platform, or a low floor, as in these conditions the agent will already be desperate for survival, so will not want to be further constrained by a treaty.
+    \item It makes a request based on HP where the threshold is within the ``weak'' or ``critical'' HP bands, as this means the agent could become critical or die as a result of following the treaty.
+    \item It would require the agent to take less food than is required to avoid entering the ``critical'' HP band.
+\end{itemize}
+If the treaty does not meet any of the above conditions, the \texttt{decision} value is then computed as:
+\[\texttt{decision} = 6 - \texttt{selfishness} + \texttt{proposerFavour} - \frac{\texttt{treatyDuration}}{3}\]
+
+This calculation is based on the following considerations:
+\begin{itemize}
+    \item Selfish agents will always be less likely to sign treaties
+    \item The proposing agent is more likely to be trusted if the receiving agent views them more favourably
+    \item Longer treaties are viewed more negatively because factors such as the receiving agent's opinion of the proposer and the agent's selfishness can change over time, meaning the same treaty could be viewed less favourably in future. Therefore, the agent should avoid constraining its actions to satisfy an agent of which it could form a negative opinion.
```

Comment:
```suggestion
    \item Longer treaties are viewed more negatively because factors such as the receiving agent's opinion of the proposer and the agent's selfishness can change over time, meaning the same treaty could be viewed less favourably in the future. Therefore, the agent should avoid constraining its actions to satisfy an agent of which it could form a negative opinion.
```